src/Microsoft.DotNet.CodeFormatting.Tests/PropertyBagTests.cs

```diff
@@ -0,0 +1,443 @@
+        }
+
+        public PropertyBag PersistAndReloadPropertyBag(PropertyBag propertyBag)
+        {
+            PropertyBag result = null;
+            string path = Path.GetTempFileName();
+            try
+            {
+                using (var writer = new FileStream(path, FileMode.OpenOrCreate, FileAccess.ReadWrite))
+                {
+                    propertyBag.SaveTo(writer, "TestProperties");
+                }
+
+                result = new PropertyBag();
+
+                using (var reader = new FileStream(path, FileMode.OpenOrCreate, FileAccess.ReadWrite))
+                {
+                    result.LoadFrom(reader);
+                }
+            }
+            finally
+            {
+                try { File.Delete(path); } catch (IOException) { }
+            }
+            return result;
+        }
+
+        [TestMethod]
+        public void PropertyBag_RemoveFromTypedPropertyBag()
+        {
+            var typedPropertyBag = new TypedPropertyBag<PropertyBag>();
+            var propertyBag = typedPropertyBag.GetProperty(PropertyBagOption);
+
+            Assert.AreEqual(1, typedPropertyBag.Count);
+
+            typedPropertyBag.SetProperty(PropertyBagOption, null);
+            Assert.AreEqual(0, typedPropertyBag.Count);
+        }
+
+        [ExpectedException(typeof(ArgumentNullException))]
+        [TestMethod]
+        public void PropertyBag_SaveToNull()
+        {
+            var propertyBag = new PropertyBag();
+            propertyBag.SaveTo((string)null, "testId");
+        }
+
+        [ExpectedException(typeof(ArgumentNullException))]
+        [TestMethod]
+        public void PropertyBag_TypedPropertyBagGetNull()
+        {
+            var propertyBag = new TestEnumPropertyBag();
+            propertyBag.GetProperty(null);
+        }
+
+        [ExpectedException(typeof(ArgumentNullException))]
+        [TestMethod]
+        public void PropertyBag_TypedPropertyBagSetNull()
+        {
+            var propertyBag = new TestEnumPropertyBag();
+            propertyBag.SetProperty(null, 0);
+        }
+
+        [TestMethod]
+        public void PropertyBag_SaveEmptyStream()
+        {
+            var propertyBag = new PropertyBag();
+            propertyBag = PersistAndReloadPropertyBag(propertyBag);
+            Assert.AreEqual(0, propertyBag.Count);
+
+            string path = Path.GetTempFileName();
+            try
+            {
+                propertyBag.SaveTo(path, "TestId");
+                propertyBag = new PropertyBag();
+                propertyBag.LoadFrom(path);
+                Assert.AreEqual(0, propertyBag.Count);
+            }
+            finally
+            {
+                if (File.Exists(path)) { File.Delete(path); }
+            }
+        }
+
+        [TestMethod]
+        public void PropertyBag_SaveNullPropertyValue()
+        {
+            var propertyBag = new PropertyBag();
+            propertyBag = PersistAndReloadPropertyBag(propertyBag);
```

Comment:
Actually testing round-trip serialize/deserialize of empty property bag. Fixed name of test to reflect this.